WordPress/src/main/java/org/wordpress/android/ui/photopicker/PhotoPickerFragment.kt

```diff
@@ -0,0 +1,701 @@
+import org.wordpress.android.util.WPMediaUtils
+import org.wordpress.android.util.WPPermissionUtils
+import org.wordpress.android.util.analytics.AnalyticsUtils
+import org.wordpress.android.util.config.TenorFeatureConfig
+import java.util.HashMap
+import javax.inject.Inject
+import kotlin.math.abs
+
+class PhotoPickerFragment : Fragment() {
+    enum class PhotoPickerIcon(private val mRequiresUploadPermission: Boolean) {
+        ANDROID_CHOOSE_PHOTO(true),
+        ANDROID_CHOOSE_VIDEO(true),
+        ANDROID_CAPTURE_PHOTO(true),
+        ANDROID_CAPTURE_VIDEO(true),
+        ANDROID_CHOOSE_PHOTO_OR_VIDEO(true),
+        WP_MEDIA(false),
+        STOCK_MEDIA(true),
+        GIF(true),
+        WP_STORIES_CAPTURE(true);
+
+        fun requiresUploadPermission(): Boolean {
+            return mRequiresUploadPermission
+        }
+    }
+
+    /*
+     * parent activity must implement this listener
+     */
+    interface PhotoPickerListener {
+        fun onPhotoPickerMediaChosen(uriList: List<Uri>)
+        fun onPhotoPickerIconClicked(icon: PhotoPickerIcon, allowMultipleSelection: Boolean)
+    }
+
+    private var actionMode: ActionMode? = null
+    private var restoreState: Parcelable? = null
+    private var listener: PhotoPickerListener? = null
+    private var lastTappedIcon: PhotoPickerIcon? = null
+    private var selectedPositions: List<Int>? = null
+    private var site: SiteModel? = null
+    private lateinit var browserType: MediaBrowserType
+
+    @Inject lateinit var tenorFeatureConfig: TenorFeatureConfig
+
+    @Inject lateinit var deviceMediaListBuilder: DeviceMediaListBuilder
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (requireActivity().application as WordPress).component().inject(this)
+        browserType = requireArguments().getSerializable(MediaBrowserActivity.ARG_BROWSER_TYPE) as MediaBrowserType
+        site = requireArguments().getSerializable(WordPress.SITE) as SiteModel
+        if (savedInstanceState != null) {
+            val savedLastTappedIconName = savedInstanceState.getString(KEY_LAST_TAPPED_ICON)
+            lastTappedIcon = savedLastTappedIconName?.let { PhotoPickerIcon.valueOf(it) }
+            if (savedInstanceState.containsKey(KEY_SELECTED_POSITIONS)) {
+                selectedPositions = savedInstanceState.getIntegerArrayList(KEY_SELECTED_POSITIONS)
+            }
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(
+                R.layout.photo_picker_fragment,
+                container,
+                false
+        )
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        if (browserType.isWPStoriesPicker) {
+            wp_stories_take_picture.visibility = View.VISIBLE
```

Comment:
no action required, just sharing 😊: I'm always a bit in doubt on which notation to use when using kotlin synthetic accessors. Found sometimes useful (to keep xml ids convention in XML only) to use the `as` like

```
import kotlinx.android.synthetic.main.photo_picker_fragment.wp_stories_take_picture as wpStoriesTakePicture
```

of course this make you lose the concise

```
import kotlinx.android.synthetic.main.photo_picker_fragment.*
```
but on the other hand it keeps code naming conventions 🤷‍♂️.

Reply:
this is an interesting point 👍 . I don't love the `*` import but I couldn't make it work with directly importing `import kotlinx.android.synthetic.main.photo_picker_fragment.wp_stories_take_picture` (AS was fighting with me). The alias seems like a useful feature. I'll try it and let you know :-)